client/src/home/edit_form/EditForm.js

```diff
@@ -0,0 +1,201 @@
+import React, { useState, useEffect } from "react";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+import DatePicker from "react-datepicker";
+import NumericInput from "react-numeric-input";
+import { useCookies } from "react-cookie";
+import { editItem } from "../../MongodbFunctions";
+import { ReactSearchAutocomplete } from 'react-search-autocomplete';
+
+import "react-datepicker/dist/react-datepicker.css";
+import "./EditForm.scss";
+
+function EditTempButton() {
+    const [formRendered, toggleFormRendered] = useState(false);
+  
+    return (
+      <div className="EditForm">
+        <Button variant="dark" onClick={() => toggleFormRendered(!formRendered)}>
+          Edit Item
+        </Button>
+        {formRendered && (
+          <EditForm closeForm={() => toggleFormRendered(false)}></EditForm>
+        )}
+      </div>
+    );
+  }
+  
+  function EditForm(props) {
+    // Input fields
+    const [itemName, setItemName] = useState("");
+    const [oldBatchId, setOldBatchId] = useState("");
+    const [newBatchId, setNewBatchId] = useState("");
+    const [lbPerHouse, setLbPerHouse] = useState(0);
+    const [weight, setWeight] = useState(0);
+    const [outByDate, setOutByDate] = useState(new Date());
+    
+    const [cookies] = useCookies(["token"]);
+    const [showWeightWarning, setWeightWarning] = useState(false);
+    const [showitemNameWarning, setitemNameWarning] = useState(false);
+    const [showbatchIdWarning, setbatchIdWarning] = useState(false);
+    // Logic for adding recommendations will need to be added
+    const [rec, setRec] = useState([]);
+    const [itemList, setItemList] = useState([]);
+  
+    useEffect(() => {
+      setItemList(props.itemList);
+    }, [props.itemList])
+  
+    return (
+      <div className="edit_form_container">
+        <Form className="edit_form">
+          <Form.Group className="input_field_container">
+            <Form.Group>
+              <Form.Label> Item Name</Form.Label>
+              <Form.Control
+                
+                
+                onChange={event => {
+                  let string = event.target.value.trim().toLowerCase();
+                  if(string.length === 0){
+                    setitemNameWarning(true);
+                  }
+                  else{ 
+                    setitemNameWarning(false);
+                  }
+                  setItemName(string);
+                }}
+                onBlur={e => {
+                    if(itemName.length === 0){
+                      setitemNameWarning(true);
+                    }
+                    else{
+                      setitemNameWarning(false);
+                    }
+                  }}
+                
+                placeholder="e.g. Apples"
```

Comment:
I think for each edit button, we need to pass in which batch the edit button is for, and all of the placeholders should be specific to that batch. So for example, whenever we put the editButton anywhere, we pass in which batch it is for and then the placeholder for name is batch.name, the placeholder of batchid is batch.id, so on and so forth.